--- a/arch/arm/kernel/setup.c
+++ b/arch/arm/kernel/setup.c
@@ -37,6 +37,10 @@
 #include <asm/mach/irq.h>
 #include <asm/mach/time.h>
 
+#if defined(CONFIG_ARCH_EP93XX)
+#include <asm/arch/ep93xx-regs.h>
+#endif
+
 #include "compat.h"
 #include "atags.h"
 
@@ -979,9 +983,16 @@ static int c_show(struct seq_file *m, vo
 	seq_puts(m, "\n");
 
 	seq_printf(m, "Hardware\t: %s\n", machine_name);
+#if defined(CONFIG_ARCH_EP93XX)
+	seq_printf(m, "Revision\t: %04x\n",
+	     *((unsigned int *)EP93XX_SYSCON_CHIP_ID) >> 28);
+	seq_printf(m, "Serial\t\t: %016x\n",
+	     *((unsigned int *)EP93XX_SECURITY_UNIQID));
+#else
 	seq_printf(m, "Revision\t: %04x\n", system_rev);
 	seq_printf(m, "Serial\t\t: %08x%08x\n",
 		   system_serial_high, system_serial_low);
+#endif
 
 	return 0;
 }
